### PR TITLE
Offline deletion tool that searches and collects stale blocks

### PR DESCRIPTION
**Pull request by kuenishi**
Supercedes #1124 . Changes since #1124 are:
- Offline delete tool:
  - Improved success count by checking delete `ok`s
  - Add an option to support old Riak 1.4 key format
  - Add `-mode(compile).`
  - Add serious warning and interactive input for running Riak process
- Select-gc tool:
  - Change name of options from/to  to start/end
  - Remove `-l` option for leeway; use start/end
  - Add `-mode(compile).`

Also, for fallbacks, I'd rather expect hinted handoff works and then delete with this tool. This is to prevent this script from being more complex, from operational mistake and from further bugs.

### CODE REVIEW

**Comment by shino** [comment]
>  for fallbacks, I'd rather expect hinted handoff works and then delete with this tool. This is to prevent this script from being more complex

Agree. Then, it's prerequisite (or recommended at least) to finish hinted handoff before running the scripts.

File: priv/tools/internal/offline_delete.erl [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    offline_delete(BitcaskDir, BlocksListFile, true, true);
+main(["--dry-run", BitcaskDir, BlocksListFile]) ->
+    offline_delete(BitcaskDir, BlocksListFile, true, false);
+main(["--old-format", BitcaskDir, BlocksListFile]) ->
+    offline_delete(BitcaskDir, BlocksListFile, false, true);
+main([BitcaskDir, BlocksListFile]) ->
+    offline_delete(BitcaskDir, BlocksListFile, false, false);
+main(_) ->
+    io:format(standard_error,
+              "options: [--dry-run][--old-format] <BitcaskDir> <BlocksListFile>~n"
```
Comment (shino) [inline priv/tools/internal/offline_delete.erl]: Space between two options?

File: priv/tools/internal/offline_delete.erl [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+%% New bitcask 1.7 format (Riak 2.0 or later)
+-define(VERSION_1, 1).
+-define(VERSION_BYTE, ?VERSION_1).
+
+make_sure(Dir) ->
+    io:format(standard_error,
+              "\033[31m[Warning]\033\[0m~n"
+              "Make sure any Riak process using '~s' is not running "
+              "or your data may corrupt.~n", [Dir]),
```
Comment (shino) [inline priv/tools/internal/offline_delete.erl]: Abs path here?

File: priv/tools/internal/select_gc_bucket.erl [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+                _State = #state{ring_size=RingSize,
+                              logger=File}
+                              ) ->
+    io:format(standard_error, "~p (~p) ~p~n", [BKey, mochihex:to_hex(UUID), ContentLength]),
+    BlockSequences = riak_cs_lfs_utils:block_sequences_for_manifest(M),
+    Count = ordsets:fold(fun({UUID1, SeqNo}, Count0) ->
+                                 BK = {B,K} = full_bkey(Bucket, dummy, UUID1, SeqNo),
+                                 VNodes = [[integer_to_list(VNode), $\t] || VNode <- vnode_ids(BK, RingSize, 3)],
+                                 %% Partitions, UUID, SeqNo
+                                 file:write(File, [VNodes, $\t,
```
Comment (shino) [inline priv/tools/internal/select_gc_bucket.erl]: Is it intentional to write double tab characters after three vnodes columns?

Comment (kuenishi) [inline priv/tools/internal/select_gc_bucket.erl]: Mistakes cannot be intentional, right?

**Comment by kuenishi** [comment]
Updated

**Comment by shino** [comment]
Works nice :smile:

**Comment by shino** [comment]
@borshop merge

**Pull request merged**